packages/mini-browser/src/browser/mini-browser-open-handler.ts

```diff
@@ -145,6 +148,9 @@ export class MiniBrowserOpenHandler extends NavigatableWidgetOpenHandler<MiniBro
     }
 
     protected resetBackground(uri: URI): MaybePromise<boolean> {
+        if (uri.toString().startsWith('localhost:') || uri.toString().startsWith('127.0.0.1')) {
```

Comment:
why these special cases? Also, when would a URI start with those prefixes, i.e. have no scheme?

Reply:
It was not possible to open any arbitrary locations before https://github.com/theia-ide/theia/pull/3451, now I can do, `Ctrl/Cmd+Shift+p` > `Open URL` > and type any locations. It won't go through the `LocationMapperService, hence there is no scheme. That's why I had to add this special handling.

![screen shot 2019-02-18 at 08 51 31](https://user-images.githubusercontent.com/1405703/52935824-bc419a80-335a-11e9-8914-aefc6dca3fdf.jpg)

![screen shot 2019-02-18 at 08 51 41](https://user-images.githubusercontent.com/1405703/52935827-bfd52180-335a-11e9-825f-aec463a00890.jpg)

Reply:
Wouldn't it be better if the command handler takes care of turning it into an http url in those cases? It looks a bit odd to have these special cases. Also if I understood correctly it should do the same if I enter any other domain name or ip address.

Reply:
Yes. we can do it differently, but do we need this `Open URL` functionality? If I remember correctly, we wanted to disable this because it won't work correctly. For instance, if I do `https://github.com/`, it won't load.

```
Refused to display 'https://github.com/' in a frame because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'none'".
```

Reply:
Yes, people were asking how to open a preview manually. So it is used for the local dev server case. But for instance in gitpod that would not be 'localhost'.

Reply:
OK, I will make some adjustments according to your feedback.